src/main/java/seedu/module/logic/commands/TagCommand.java

```diff
@@ -74,27 +73,24 @@ public CommandResult execute(Model model) throws CommandException {
     }
 
     /**
-     * Add tag to set of old tags, duplicated tag will not be added
+     * Add tags to set of old tags, duplicated tag will not be added
      *
      * @param oldTags set of old tags
-     * @param newTag new Tag to be added
+     * @param newTags new Tags to be added
      * @return set of new tags
      */
-    private Set<Tag> addTags(Set<Tag> oldTags, Tag newTag) {
-        Set<Tag> newTags = new HashSet<>(oldTags);
-        boolean isEqual = false;
-        for (Tag item : oldTags) {
-            if (item.equals(newTag)) {
-                isEqual = true;
-                break;
-            }
-        }
-        if (isEqual) {
-            return newTags;
-        } else {
-            newTags.add(newTag);
-            return newTags;
-        }
+    private Set<Tag> addTags(Set<Tag> oldTags, Set<Tag> newTags) {
+        Set<Tag> resultTags = new HashSet<>(oldTags);
+        resultTags.addAll(newTags);
+        return resultTags;
+    }
+
+    /**
+     * Sets {@code tags} to this object's {@code tags}.
+     * A defensive copy of {@code tags} is used internally.
+     */
+    public void setTags(Set<Tag> tags) {
+        this.tags = (tags != null) ? new HashSet<>(tags) : null;
```

Comment:
It might be more "visual" to write this in a few more lines. But no big deal lgtm.

Reply:
Updated